.github/workflows/build-docs.yml

```diff
@@ -9,23 +9,46 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.9
-      uses: actions/setup-python@v2
-      with:
-        python-version: 3.9
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements.txt
-        pip install -r requirements_dev.txt
-        pip install -e .
-    - name: Build documentation
-      run: |
-        python -m sphinx -T -b html -d docs/_build/doctrees -D language=en docs docs/_build/html -n -W
-    - uses: actions/upload-artifact@v2
-      name: Upload docs as artifact
-      with:
-        name: docs-html
-        path: './docs/_build/html'
-        if-no-files-found: error
+      - uses: actions/checkout@v2
+
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Get full Python version
+        id: full-python-version
+        run: echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
+
+      - name: Bootstrap poetry
+        run: |
+          curl -sL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python - -y
```

Comment:
Why not use `pip install poetry`?

Reply:
You're right that `pip install poetry` _should_ work as well. I see three reasons to use the installer script though:
1. [Poetry is using the installer script in their own GitHub Actions workflows](https://github.com/python-poetry/poetry/blob/master/.github/workflows/main.yml#L51).
2. [Using the installer script is the recommended way to install Poetry](https://python-poetry.org/docs/#installing-with-pip).
3. When Poetry is installed via `pip`, `pip` will resolve Poetry's dependencies and install their latest versions according to the version specifiers. While this _should_ work as well, it is not fully deterministic when new versions of Poetry's dependencies are released. When Poetry is installed via the installer script, [its dependencies are vendored](https://github.com/python-poetry/poetry/blob/master/sonnet) and (I assume) this specific set of packages is tested to be working.

Something worth thinking about might be to install a specific version of Poetry using the installer script (e.g. `--version 1.1.7`) to make the installed Poetry version deterministic, too. I wonder why Poetry itself in their GitHub Actions workflows is installing the latest version (whatever it is). What do you think?

Reply:
> Poetry is using the installer script in their own GitHub Actions workflows.

That's actually really strange. They should use branch version 🤔 

Other than that it sounds reasonable.

Reply:
> That's actually really strange. They should use branch version :thinking:

:thinking: I wonder if it were weird to do that because they would be using their own project at the state of that branch to install itself. Let's say something got messed up in that branch so that Poetry could not install dependencies anymore, then the CI workflow would fail before the test suite was run, so a detailed error analysis is more difficult. By using the installer script, they're using the latest release which should have passed all tests.